### PR TITLE
add deprecation to nodejs 16

### PR DESCRIPTION
**Pull request by Jdubrick**
### What does this PR do?:
_Summarize the changes. Are any stacks or samples added or updated?_

This PR adds the `Deprecated` tag to stacks with the `Nodejs 16` image. It also adds these deprecated stacks to the `renovate.json` exclusion list and bumps the default version for all changed stacks up by 1 so the default version is not a deprecated stack.

### Which issue(s) this PR fixes:
_Link to github issue(s)_
fixes https://github.com/devfile/api/issues/1364

### PR acceptance criteria:

- [ ] Contributing guide
_Have you read the [devfile registry contributing guide](https://github.com/devfile/registry/blob/main/CONTRIBUTING.md) and followed its instructions?_
- [ ] Test automation
_Does this repository's tests pass with your changes?_
- [ ] Documentation
_Does any documentation need to be updated with your changes?_
- [ ] Check Tools Provider
_Have you tested the changes with existing tools, i.e. Odo, Che, Console? (See [devfile registry contributing guide](https://github.com/devfile/registry/blob/main/CONTRIBUTING.md) on how to test changes)_

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -44,7 +44,14 @@
...
     "stacks/go/2.0.0/**",
-    "stacks/go/2.1.0/**"
+    "stacks/go/2.1.0/**",
+    "stacks/nodejs/2.1.1/**",
+    "stacks/nodejs-angular/2.0.2/**",
+    "stacks/nodejs-nextjs/1.0.3/**",
+    "stacks/nodejs-nuxtjs/1.0.3/**",
+    "stacks/nodejs-react/2.0.2/**",
+    "stacks/nodejs-svelte/1.0.3/**",
+    "stacks/nodejs-vue/1.0.3/**"
```
Comment (michael-valdron) [inline renovate.json:54]: Think a typo in the intended `nodejs-vue` stack version:
```suggestion
    "stacks/nodejs-vue/1.0.2/**"
```

**Review by michael-valdron — CHANGES_REQUESTED** [review]

File: renovate.json (hunk 1 above)
Comment (Jdubrick) [reply renovate.json:54]: Oops my bad, I'll correct that real quick thank you!

**Comment by Jdubrick** [comment]
/hold

This is going to need to be merged after the nodejs issue is fixed https://github.com/devfile/api/issues/1484

**Review by thepetk — COMMENTED** [review]
I think based on what we have discussed for https://github.com/devfile/api/issues/1482 for the stacks we update the default version we can make the new default their latest version. WDYT?

**Comment by Jdubrick** [comment]
> I think based on what we have discussed for [devfile/api#1482](https://github.com/devfile/api/issues/1482) for the stacks we update the default version we can make the new default their latest version. WDYT?

That's okay with me, I know I just bumped them all up 1 but can update it to bump them all to latest. I'm going to also mark the issue as blocked since it isn't going to pass PR checks with the nodejs icon issue

cc @thepetk

**Comment by michael-valdron** [comment]
Just need https://github.com/devfile/registry/pull/338 merged after hearing back from the stack owner before the builds will resume working again

**Comment by michael-valdron** [comment]
/unhold

https://github.com/devfile/registry/pull/338 should have unblocked this PR.

**Comment by Jdubrick** [comment]
Will re run tests in a little bit, seems like the issue is still occuring but I don't know maybe it hasn't updated yet to CI. https://github.com/devfile/registry/actions/runs/8348517850/job/22941030162?pr=337

**Comment by michael-valdron** [comment]
> Will re run tests in a little bit, seems like the issue is still occuring but I don't know maybe it hasn't updated yet to CI. https://github.com/devfile/registry/actions/runs/8348517850/job/22941030162?pr=337

Most likely, `ci/prow/v4.13-images` is passing now.

**Comment by michael-valdron** [comment]
@lholmquist Need your quick review on the changes done to the `nodejs` stack.

FYI @Jdubrick

**Review by lholmquist — APPROVED** [review]

**Review by michael-valdron — CHANGES_REQUESTED** [review]
@Jdubrick Just realized you haven't rebased since https://github.com/devfile/registry/pull/338 was merged :see_no_evil:

So your version of the code is still got the broken nodejs icon links.

**Comment by Jdubrick** [comment]
> @Jdubrick Just realized you haven't rebased since #338 was merged 🙈
> 
> So your version of the code is still got the broken nodejs icon links.

@michael-valdron i was trying to figure out why the builds were still failing, thank you. Just rebased so hopefully this fixes the issue!

**Comment by Jdubrick** [comment]
@michael-valdron looks like the rebase and nodejs changes allowed checks to pass

File: stacks/nodejs-vue/stack.yaml [hunk 2]
```diff
@@ -6,6 +6,6 @@ displayName: Vue
 icon: https://raw.githubusercontent.com/devfile-samples/devfile-stack-icons/main/vue.svg
 versions:
   - version: 1.0.2
-    default: true # should have one and only one default version
   - version: 1.1.0
+    default: true # should have one and only one default version
   - version: 1.2.0
```
Comment (thepetk) [inline stacks/nodejs-vue/stack.yaml:11]: Shouldn't `1.2.0` be the latest?

File: stacks/nodejs-react/stack.yaml [hunk 3]
```diff
@@ -6,6 +6,6 @@ displayName: React
 icon: https://raw.githubusercontent.com/devfile-samples/devfile-stack-icons/main/react.svg
 versions:
   - version: 2.0.2
-    default: true # should have one and only one default version
   - version: 2.1.0
+    default: true # should have one and only one default version
   - version: 2.2.0
```
Comment (thepetk) [inline stacks/nodejs-react/stack.yaml:11]: Shouldn't be `2.2.0` be the latest

File: stacks/nodejs-nuxtjs/stack.yaml [hunk 4]
```diff
@@ -6,6 +6,6 @@ displayName: Nuxt.js
 icon: https://raw.githubusercontent.com/devfile-samples/devfile-stack-icons/main/nuxt-js.svg
 versions:
   - version: 1.0.3
-    default: true # should have one and only one default version
   - version: 1.1.0
+    default: true # should have one and only one default version
   - version: 1.2.0
```
Comment (thepetk) [inline stacks/nodejs-nuxtjs/stack.yaml:11]: Shouldn't be the `1.2.0` the latest?

File: stacks/nodejs-vue/stack.yaml [hunk 5]
```diff
@@ -6,6 +6,6 @@ displayName: Vue
 icon: https://raw.githubusercontent.com/devfile-samples/devfile-stack-icons/main/vue.svg
 versions:
   - version: 1.0.2
-    default: true # should have one and only one default version
   - version: 1.1.0
+    default: true # should have one and only one default version
```
Comment (thepetk) [inline stacks/nodejs-vue/stack.yaml:10]: Shouldn't it be the `1.2.0` one?

**Comment by Jdubrick** [comment]
@thepetk in response to your reviews the defaults were just bumped up 1 version from the one that was deprecated in this case. This was done before any discussion surrounding if we should keep the default updated to the latest version. wdyt?

**Comment by thepetk** [comment]
> @thepetk in response to your reviews the defaults were just bumped up 1 version from the one that was deprecated in this case. This was done before any discussion surrounding if we should keep the default updated to the latest version. wdyt?

I'd say in order to avoid back & forth for this PR we can move the default to latest with a separate PR!

**Review by michael-valdron — APPROVED** [review]
/lgtm

**Pull request merged**